src/cml.js

```diff
@@ -303,6 +303,7 @@ class CML {
         }
       }
 
+      await exec(`git fetch origin ${sha}`);
```

Comment:
Please revert this as it was. This approach was actually discarded.
We have to test also in GL and BB.

In any case it should be 
```js
await exec(`git fetch ${remote} ${sha}`);
```

But again please revert it 🙏

Reply:
```suggestion
      await exec(`git fetch ${remote} ${sha}`);
```

Reply:
> _Please revert this as it was._

Do you mean _don't use `git fetch` at all_ or just _revert it to `git fetch` without arguments as in the https://github.com/iterative/cml/pull/624/commits/392b6e11d19007664417124e9f6b75127de8e676 commit above?

Reply:
> _This approach was actually discarded. We have to test also in GL and BB._

May I know why? I haven't tested it on GitLab nor Bitbucket, but I assumed that it would work. 🐒

Reply:
> May I know why? I haven't tested it on GitLab nor Bitbucket, but I assumed that it would work. 

Every CI gives you the repo in a different condition. Said that... I do not remember why but if you dive in the original PR you would be able to see this approach even using js git

Reply:
> _Every CI gives you the repo in a different condition._

So true! 🙃 

> _Said that... I do not remember why but if you dive in the original PR you would be able to see this approach even using js git_

I'd test it as-is also on GitLab and see what happens. Running `git fetch` with a fixed SHA shouldn't have any negative impact on the repository and would avoid breaking with GitHub shallow clones.

Reply:
I have remembered. Originally it was doing fetch to have all the history and check if the cmlpr branch already existed. Was indeed on of the very first operations. Later on I started to use the API instead and fetch at branch my suffice... but I would check also

Reply:
#### Without `fetch`
```yaml
on: pull_request
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - run: |
          sudo npm config set user 0
          sudo npm install --global https://github.com/iterative/cml#fix-pull-request-references
          date > file
          cml-pr file     
        env:
          GITHUB_TOKEN: ${{ github.token }}
```
  
#### With `fetch`

```yaml
on: pull_request
jobs:
  test:
    runs-on: ubuntu-latest
    steps:
      - uses: actions/checkout@v2
      - run: |
          sudo npm config set user 0
          sudo npm install --global https://github.com/iterative/cml#fix-pull-request-references
          git fetch origin ${{ github.event.pull_request.head.sha }}
          date > file
          cml-pr file     
        env:
          GITHUB_TOKEN: ${{ github.token }}
```